src/components/SaveAsTemplateModal.tsx

```diff
@@ -0,0 +1,115 @@
+import ReactModal from "react-modal";
+import { FormattedMessage } from "react-intl";
+import { useCreatePrivateTemplate } from "../queries/template-queries";
+import {
+  CampaignContent,
+  TemplateContent,
+} from "../abstractions/domain/content";
+
+interface SaveAsTemplateModalProps {
+  onClose: () => void;
+  content: TemplateContent;
+}
+
+export const SaveAsTemplateModal = ({
+  onClose,
+  content,
+}: SaveAsTemplateModalProps) => {
+  const {
+    mutate: createPrivateTemplate,
+    isLoading,
+    isSuccess,
+  } = useCreatePrivateTemplate();
+
+  const createTemplate = () => {
+    createPrivateTemplate(content);
+  };
+
+  const successContent = (
+    <>
+      <p>
+        <FormattedMessage id="new_template_has_been_saved"></FormattedMessage>
+      </p>
+      <div className="form-request">
+        <div className="container-buttons">
+          <button
+            type="button"
+            name="ok"
+            className="dp-button button-medium primary-green"
+            onClick={onClose}
+          >
+            <FormattedMessage id="accept"></FormattedMessage>
+          </button>
+        </div>
+      </div>
+    </>
+  );
+
+  const formContent = (
+    <>
+      <p>
+        <FormattedMessage id="new_template_description"></FormattedMessage>
+      </p>
+      <form action="#" className="form-request">
+        <fieldset>
+          <legend>Guardar como Plantilla privada</legend>
+          <ul className="field-group">
+            <li className="field-item">
+              <label htmlFor="name">
+                <FormattedMessage id="new_template_label"></FormattedMessage>
+              </label>
+              <input
+                name="name"
+                id="name"
+                defaultValue={content.templateName}
+              />
+            </li>
+          </ul>
+        </fieldset>
+        <div className="container-buttons">
+          <button
+            type="button"
+            name="cancel"
+            className="dp-button button-medium primary-grey"
+            onClick={onClose}
+          >
+            <FormattedMessage id="cancel"></FormattedMessage>
+          </button>
+          <button
+            type="button"
+            name="submit"
+            onClick={createTemplate}
```

Comment:
If I am not wrong, `createTemplate` should be executed by a form's event, not by a button's event.

Reply:
Yes, now is creating template in onSubmit event.

Reply:
👍